specs/template.go

```diff
@@ -0,0 +1,106 @@
+package specs
+
+import (
+	"strings"
+
+	"github.com/zclconf/go-cty/cty"
+)
+
+const (
+	// TemplateOpen tag
+	TemplateOpen = "{{"
+	// TemplateClose tag
+	TemplateClose = "}}"
+	// ReferenceDelimiter represents the value resource reference delimiter
+	ReferenceDelimiter = ":"
+	// PathDelimiter represents the path reference delimiter
+	PathDelimiter = "."
+
```

Comment:
Shouldn’t we add UserDefinedFunctionOpen = "(" and UserDefinedFunctionClose = ")" over here

Reply:
The custom defined functions are pattern matched using regex, the pattern is stored in `FunctionPattern`.